Druid/DRSmallIntegerPrimitiveSpecialisation.class.st

```diff
@@ -0,0 +1,64 @@
+Class {
+	#name : #DRSmallIntegerPrimitiveSpecialisation,
+	#superclass : #DRPrimitiveSpecialisation,
+	#category : #'Druid-CompilerCompiler'
+}
+
+{ #category : #accessing }
+DRSmallIntegerPrimitiveSpecialisation class >> pragmaName [
+
+	^ #smallInteger
+]
+
+{ #category : #api }
+DRSmallIntegerPrimitiveSpecialisation >> addInitialConstraint: anIR [
+
+	anIR initialConstraint: self initialConstraint
```

Comment:
I wonder why do we need two hooks

Reply:
What so you refer with "two hooks"?

Reply:
Well, there is customiseFor:, specializeFor: and this one